### PR TITLE
OCP4 STIG: deduplicate audit rules

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- The OCP4 STIG profile was selecting two different sets of rules that create audit rules for syscalls with different approaches. There are rules that check for and remediate audit rules for a:
  - single syscall
  -  multiple syscalls
-  This PR changes the STIG profile to use the rules that handle multiple syscalls.

#### Rationale:
- As the audit rules dealing with syscall arguments need to be in a specific order, it is safer to use these rules that handle multiple syscalls at a time.

#### Review Hints:

- Apply the `rhcos4-stig` profile and check that there are no duplicate / equivalent audit rules.

### CODE REVIEW

**Comment by rhmdnd** [comment]
Does this need a rebase to pickup fixes for the STIG e2e failure?

```
helpers.go:872: E2E-ERROR: The e2e YAML for rule 'e2e-stig-oauth-or-oauthclient-inactivity-timeout' is malformed: {<nil> <nil> <nil>} . Got error: couldn't parse the result as string or map
```

File: controls/stig_ocp4.yml [hunk 1]
```diff
@@ -914,12 +911,6 @@ controls:
   - audit_rules_privileged_commands_usermod
   - audit_rules_sudoers
   - audit_rules_sudoers_d
-  - audit_rules_unsuccessful_file_modification_creat
-  - audit_rules_unsuccessful_file_modification_ftruncate
-  - audit_rules_unsuccessful_file_modification_open
-  - audit_rules_unsuccessful_file_modification_open_by_handle_at
-  - audit_rules_unsuccessful_file_modification_openat
-  - audit_rules_unsuccessful_file_modification_truncate
   - audit_rules_usergroup_modification_group
```
Comment (xiaojiey) [inline controls/stig_ocp4.yml:914]: Do we need to remove the below 4 rules? I can see they are duplicated with lines here at https://github.com/ComplianceAsCode/content/blob/7dc7e06c88204afdada5658928f1642151847bfa/control/stig_ocp4.yml#L967-L970.
I might be wrong as I don't know the background of this PR.

```
  - audit_rules_usergroup_modification_group
  - audit_rules_usergroup_modification_gshadow
  - audit_rules_usergroup_modification_passwd
  - audit_rules_usergroup_modification_shadow
```

File: controls/stig_ocp4.yml [hunk 2]
```diff
@@ -81,7 +79,6 @@ controls:
   title: OpenShift must generate audit rules to capture account related actions.
   rules:
   - audit_rules_sysadmin_actions
-  - audit_rules_usergroup_modification
```
Comment (xiaojiey) [inline controls/stig_ocp4.yml:84]: Is this rule removed on purpose? The rule audit-rules-usergroup-modification will be removed from the rhcos4-stig profile with this PR.
```
 % oc get profile rhcos4-stig -o=jsonpath={.rules} | jq -r | grep audit-rules-usergroup-modification
  "rhcos4-audit-rules-usergroup-modification",
  "rhcos4-audit-rules-usergroup-modification-group",
  "rhcos4-audit-rules-usergroup-modification-gshadow",
  "rhcos4-audit-rules-usergroup-modification-opasswd",
  "rhcos4-audit-rules-usergroup-modification-passwd",
  "rhcos4-audit-rules-usergroup-modification-shadow",
% oc get profile upstream-rhcos4-stig -o=jsonpath={.rules} | jq -r | grep audit-rules-usergroup-modification
  "upstream-rhcos4-audit-rules-usergroup-modification-group",
  "upstream-rhcos4-audit-rules-usergroup-modification-gshadow",
  "upstream-rhcos4-audit-rules-usergroup-modification-opasswd",
  "upstream-rhcos4-audit-rules-usergroup-modification-passwd",
  "upstream-rhcos4-audit-rules-usergroup-modification-shadow",
```

Comment (yuumasato) [reply controls/stig_ocp4.yml:84]: Yes, rule [audit_rules_usergroup_modification](https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/auditing/auditd_configure_rules/audit_rules_usergroup_modification/rule.yml) adds the same audit rules as the individual ones.
- -w /etc/group -p wa -k audit_rules_usergroup_modification
- -w /etc/passwd -p wa -k audit_rules_usergroup_modification
- -w /etc/gshadow -p wa -k audit_rules_usergroup_modification
- -w /etc/shadow -p wa -k audit_rules_usergroup_modification
- -w /etc/security/opasswd -p wa -k audit_rules_usergroup_modification

This was causing these rules to be duplicated after remediation is applied.
The logs below are from a profile before this patch. 
```console
$ oc get rems | grep usergroup
rhcos4-stig-master-audit-rules-usergroup-modification                             Applied
rhcos4-stig-master-audit-rules-usergroup-modification-group                       Applied
rhcos4-stig-master-audit-rules-usergroup-modification-gshadow                     Applied
rhcos4-stig-master-audit-rules-usergroup-modification-opasswd                     Applied
rhcos4-stig-master-audit-rules-usergroup-modification-passwd                      Applied
rhcos4-stig-master-audit-rules-usergroup-modification-shadow                      Applied
```

```
sh-5.1# ls -l /etc/audit/rules.d/
total 32
-rw-r--r--. 1 root root  58 Aug 13 13:38 30-etc_group_usergroup_modification.rules
-rw-r--r--. 1 root root  60 Aug 13 13:38 30-etc_gshadow_usergroup_modification.rules
-rw-r--r--. 1 root root  59 Aug 13 13:38 30-etc_passwd_usergroup_modification.rules
-rw-r--r--. 1 root root  69 Aug 13 13:38 30-etc_security_opasswd_usergroup_modification.rules
-rw-r--r--. 1 root root  59 Aug 13 13:38 30-etc_shadow_usergroup_modification.rules
-rw-r--r--. 1 root root 305 Aug 13 13:38 75-audit_rules_usergroup_modification.rules
-rw-------. 1 root root 244 Aug 13 12:18 audit.rules
-rw-r--r--. 1 root root 572 Aug 13 13:38 mco-audit-quiet-containers.rules
```

```console
sh-5.1# chroot /host    
sh-5.1# cat /etc/audit/rules.d/audit.rules
## This file is automatically generated from /etc/audit/rules.d
-D
-b 8192
-f 1
-w /etc/group -p wa -k audit_rules_usergroup_modification
-w /etc/gshadow -p wa -k audit_rules_usergroup_modification
-w /etc/passwd -p wa -k audit_rules_usergroup_modification
-w /etc/security/opasswd -p wa -k audit_rules_usergroup_modification
-w /etc/shadow -p wa -k audit_rules_usergroup_modification
-w /etc/group -p wa -k audit_rules_usergroup_modification
-w /etc/passwd -p wa -k audit_rules_usergroup_modification
-w /etc/gshadow -p wa -k audit_rules_usergroup_modification
-w /etc/shadow -p wa -k audit_rules_usergroup_modification
-w /etc/security/opasswd -p wa -k audit_rules_usergroup_modification
--backlog_wait_time 60000
-a exclude,always -F msgtype=NETFILTER_CFG
-a exclude,always -F msgtype=ANOM_PROMISCUOUS


```

File: controls/stig_ocp4.yml (hunk 1 above)
Comment (yuumasato) [reply controls/stig_ocp4.yml:914]: No, there is no need to remove these rules.
They are part of both STIG controls:
- https://stigaview.com/products/ocp/v1r1/CNTR-OS-000930/
- https://stigaview.com/products/ocp/v1r1/CNTR-OS-000950/

And selecting the same rule in multiple controls is fine. They are added only once to the profile.

**Comment by yuumasato** [comment]
Rebased, :)

**Comment by yuumasato** [comment]
@xiaojiey @rhmdnd This should be good to go

**Review by rhmdnd — APPROVED** [review]
/lgtm

**Pull request merged**